pages/organizations.js

```diff
@@ -0,0 +1,223 @@
+        ) : (
+          { ...user, role: 'manager' }
+        )
+      })
+      setMembers(checkForSingleOwner(membersWithRoles))
+    }
+  }, [users])
+
+  if (loading) {
+    return <LoadingState />
+  }
+  if (!authenticatedUser.loggedIn) {
+    return <NotLoggedIn />
+  }
+
+  /**
+ * Adds a new member with the role of manager.
+ * @param {object} user - object containing all data listed on the user table.
+ */
+  const addManager = (user) => setMembers([...members, { ...user, role: 'manager' }])
+
+  /**
+ * Adds a new parameter to owner if they are the only owner in the list.
+ * @param {array} members - object containing all data listed on the user table.
+ */
+  const checkForSingleOwner = (members) => {
+    const owners = members.filter(member => member.role === 'owner')
+    const singleOwner = owners.length === 1 ? owners.shift() : null
+    return singleOwner ? members.map(member => member.osm_id === singleOwner.osm_id ? { ...member, singleOwner: true } : member) : members
+  }
+
+  /**
+ * Updates member state with newly selected role.
+ * @param {object} user - object containing all user data from the member table.
+ * @param {string} role - new role selected from dropdown oneOf ['owner', 'manager', 'none'].
+ */
+  const setStatus = (user, role) => {
+    if (user.osm_id.toString() === props.authenticatedUser.account.id) {
+      findInitialRole('owners', user) && setWarning('Are you sure you want to change your role? You will lose the ability to add new owners in the future.')
+      role === 'none' && setWarning("Are you sure you want to remove yourself? You won't be able to add yourself back in.")
+    }
+    const updatedMembers = members.map(member => member.osm_id === user.osm_id ? { ...member, role } : member)
+    setMembers(checkForSingleOwner(updatedMembers))
+  }
+
+  /**
+ * Identifies role of user on initial load - prior to any updates.
+ * @param {string} role - role from  oneOf ['owners', 'managers'].
+ * @param {object} member - object containing all user data from the member table.
+ */
+  const findInitialRole = (role, member) => (
+    organization.organization[role].find(orgUser => orgUser.osm_id === member.osm_id)
+  )
+
+  /**
+ * Makes request for each member to change status. Since the API adds and removes members one
+ * at a time, each request is submitted separately rather than in batches.
+ */
+  const submitChanges = () => {
+    members.forEach(member => {
+      switch (member.role) {
+        case 'owner':
+          if (findInitialRole('managers', member)) {
+            props.removeManager(member.osm_id)
+          }
+          props.addOwner(member.osm_id)
+          break
+        case 'manager':
+          if (findInitialRole('owners', member)) {
+            props.removeOwner(member.osm_id)
+          }
+          props.addManager(member.osm_id)
+          break
+        case 'none':
+          props.removeManager(member.osm_id)
+          props.removeOwner(member.osm_id)
```

Comment:
I am not sure if this matters at all, but my IDE is flagging warnings here re: `Promise returned from xxxx is ignored`
<img width="429" alt="Screen Shot 2020-10-15 at 2 26 49 PM" src="https://user-images.githubusercontent.com/766758/96184979-77d31d00-0ef6-11eb-8653-8f6edf16fa43.png">

Reply:
Could this be a result of the requests that don't execute because of duplicates?